com.sap.cloud.lm.sl.cf.web/src/main/java/com/sap/cloud/lm/sl/cf/web/configuration/UAAClientConfiguration.java

```diff
@@ -51,7 +51,7 @@ private URL readTokenEndpoint(URL targetURL) {
     protected Map<String, Object> getControllerInfo(URL targetURL) {
         String infoURL = targetURL.toString() + CONTROLLER_INFO_ENDPOINT;
         ResponseEntity<String> infoResponse = new RestTemplate().getForEntity(infoURL, String.class);
-        if (infoResponse == null || infoResponse.getBody() == null) {
+        if (infoResponse.getBody() == null) {
```

Comment:
Why remove the null check?

Reply:
Because getForEntity() can't return null value and sonar reports that. Spring do check for null value and throw IllegalStateException if it is null.